### PR TITLE
settings backend request

### PR DESCRIPTION
**Pull request by salmanja**
## Describe your changes

Briefly describe the changes you made and their purpose.

## Issue number

Mention the issue number(s) this PR addresses (e.g., #123).

## Please ensure all items are checked off before requesting a review:

- [x] I deployed the code locally.
- [x] I have performed a self-review of my code.
- [x] I have included the issue # in the PR.
- [x] I have labelled the PR correctly.
- [x] The issue I am working on is assigned to me.
- [x] I didn't use any hardcoded values (otherwise it will not scale, and will make it difficult to maintain consistency across the application).
- [x] I made sure font sizes, color choices etc are all referenced from the theme.
- [x] My PR is granular and targeted to one specific feature.
- [x] I took a screenshot or a video and attached to this PR if there is a UI change.

## Summary by CodeRabbit

- **New Features**
	- Enhanced user data fetching and updating logic in the ProfileForm component.
	- Added a confirmation modal for saving changes.
	- Introduced an error modal for displaying validation errors.
	- Added a Delete Account Confirmation modal for account deletion.
	- User information can now be accessed without authentication for certain routes.

- **Bug Fixes**
	- Improved error handling for user data retrieval and updates with clearer error messages.

- **Style**
	- Minor formatting adjustments for improved code readability.

### CODE REVIEW

File: Clients/src/presentation/pages/SettingsPage/Profile/index.tsx [hunk 1]
```diff
@@ -81,10 +77,18 @@ const ProfileForm: React.FC = () => {
...
+        const userId = (await localStorage.getItem("userId")) || "1";
+        if (!userId) {
+          throw new Error("User ID not found in local storage");
+        }
+
+        const response = await fetch(`http://localhost:3000/users/${userId}`);
+        if (!response) {
+          throw new Error("failed to fetch user data");
+        }
+        const user = await response.json();
```
[the remark this answers is not in the document]
Comment (MuhammadKhalilzadeh) [reply Clients/src/presentation/pages/SettingsPage/Profile/index.tsx:91]: Please consider adding these modifications

**Comment by salmanja** [comment]
https://github.com/user-attachments/assets/a1fdec84-2df3-4f8d-bda2-bc922ab11cc4

@MuhammadKhalilzadeh suggestions completed

**Comment by salmanja** [comment]
> When the user clicks on Retry, he has the click on Cancel to go to the main dashboard again. Is there a reason for this?

I will fix the errorModalOpen in the onProceed logic

**Comment by salmanja** [comment]
> When the user clicks on Retry, he has the click on Cancel to go to the main dashboard again. Is there a reason for this?


https://github.com/user-attachments/assets/629b79f6-7a0b-4de8-876f-1d965fd8caa1

**Comment by gorkem-bwl** [comment]
After retry, the page shouldn't reload but stay the same. The errors are already there to fix.

**Comment by salmanja** [comment]
> After retry, the page shouldn't reload but stay the same. The errors are already there to fix.


https://github.com/user-attachments/assets/863ad699-3bd7-487d-8aa8-548272c39020

**Comment by gorkem-bwl** [comment]
> > After retry, the page shouldn't reload but stay the same. The errors are already there to fix.
> 
>  Screen.Recording.2024-12-16.at.5.39.48.PM.mov

Thanks, that works.

**Comment by MuhammadKhalilzadeh** [comment]
> > After retry, the page shouldn't reload but stay the same. The errors are already there to fix.
> 
>  Screen.Recording.2024-12-16.at.5.39.48.PM.mov

Hi @salmanja 
I hope you're having a good time

Thank you for applying modifications 🙏🏻
Currently, this is not how the Settings look like!
Please sync your repo.
Make sure about not having any conflicts,
Then, please upload an other screen shot or recording.

Thank you in advance

**Comment by salmanja** [comment]
> > > After retry, the page shouldn't reload but stay the same. The errors are already there to fix.
> > 
> > 
> > Screen.Recording.2024-12-16.at.5.39.48.PM.mov
> 
> Hi @salmanja I hope you're having a good time
> 
> Thank you for applying modifications 🙏🏻 Currently, this is not how the Settings look like! Please sync your repo. Make sure about not having any conflicts, Then, please upload an other screen shot or recording.
> 
> Thank you in advance


https://github.com/user-attachments/assets/65542784-a64d-463b-b401-aaf4156b5547

Screenshot 2024-12-17 at 10 50 23 AM

UI is synced and the retry button in error modal works again

**Pull request merged**